### PR TITLE
Plugin integration tests: add oss auth helper

### PR DESCRIPTION
**Pull request by hugoShaka**
Followup on: https://github.com/gravitational/teleport/pull/38175

This PR adds the OSS auth helper for the plugin integration tests.
This helper is responsible for starting the auth server and crafting client credentials used by the test suite.

I will send another PR to add a similar helper in the teleport.e repo.

Once this is merged, I will be able to send the PRs migrating tests for each plugin to the new integration test suite.

### CODE REVIEW

File: integrations/lib/testing/integration/authhelper.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+		AcceptedUsage: authServer.AcceptedUsage,
+	})
+	require.NoError(t, err)
+	a.server = server
+
+	t.Cleanup(func() {
+		require.NoError(t, server.Close())
+	})
+
+	authClient, err := server.NewClient(libauth.TestAdmin())
```
Comment (zmb3) [inline integrations/lib/testing/integration/authhelper.go:77]: Should the client also be closed on cleanup?

File: integrations/lib/testing/integration/authhelper.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+// OSSAuthHelper implements the AuthHelper interface.
+// It starts an OSS Auth server and exposes the functions required for plugin
+// integration tests to build teleport clients for each user/plugin/bot.
+type OSSAuthHelper struct {
+	server *libauth.TestTLSServer
+	dir    string
+}
+
+// StartServer implements the AuthHelper interface.
```
Comment (zmb3) [inline integrations/lib/testing/integration/authhelper.go:48]: Can we super clear in the godoc what (if anything) needs to be manually closed and what is automatically cleaned up?

**Review by zmb3 — APPROVED** [review]

File: integrations/lib/testing/integration/authhelper.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
```
Comment (marcoandredinis) [inline integrations/lib/testing/integration/authhelper.go:3]: :nail_care: 
```suggestion
 * Copyright (C) 2024  Gravitational, Inc.
```

File: integrations/lib/testing/integration/authhelper.go [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+	})
+
+	authClient, err := server.NewClient(libauth.TestAdmin())
+	require.NoError(t, err)
+
+	return authClient.APIClient
+}
+
+// ServerAddr implements the AuthHelper interface.
+// It returns the server address, including the port.
```
Comment (marcoandredinis) [inline integrations/lib/testing/integration/authhelper.go:84]: An example would be good as well.

File: integrations/lib/testing/integration/authhelper.go [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+func (a *OSSAuthHelper) ServerAddr() string {
+	return a.server.Addr().String()
+}
+
+// getUserCerts generates a private key for the user and has the auth sign a TLS and an SSH certificate.
+// The function returns three values:
+// - the PEM-encoded private key
+// - the Authorized-key formatted SSH cert
+// - the PEM-encoded TLS cert
+func (a *OSSAuthHelper) getUserCerts(t *testing.T, user types.User) ([]byte, []byte, []byte) {
```
Comment (marcoandredinis) [inline integrations/lib/testing/integration/authhelper.go:94]: Can we have a new type that includes all three fields?
Returning 3 `[]byte` might create some confusion during development

**Review by marcoandredinis — APPROVED** [review]

File: integrations/lib/testing/integration/authhelper.go (hunk 5 above)
Comment (tigrato) [inline integrations/lib/testing/integration/authhelper.go:94]:
```suggestion
func (a *OSSAuthHelper) getUserCerts(t *testing.T, user types.User) (pemKey, sshCert, tlsCert []byte) {
```

or 

```suggestion
type userCertts struct{
ssh []byte
tls []byte
pemKey []byte
}

func (a *OSSAuthHelper) getUserCerts(t *testing.T, user types.User) (userCertts) {
```

**Pull request merged**